File: reference/spring-boot-java-gradle-hexagonal/src/main/java/com/example/demo/adapter/outbound/persistence/WidgetJpaRepository.java
package com.example.demo.adapter.outbound.persistence;

import org.springframework.data.jpa.repository.JpaRepository;

/** Spring Data's view of the table. Package-private: only the adapter beside it may use it. */
interface WidgetJpaRepository extends JpaRepository<WidgetRow, Long> {

    boolean existsByName(String name);
}

File: reference/spring-boot-kotlin-gradle-hexagonal/src/main/kotlin/com/example/demo/adapter/outbound/persistence/WidgetJpaRepository.kt
package com.example.demo.adapter.outbound.persistence

import org.springframework.data.jpa.repository.JpaRepository

/** Spring Data's view of the table. `internal`: only the adapter beside it may use it. */
internal interface WidgetJpaRepository : JpaRepository<WidgetRow, Long> {
    fun existsByName(name: String): Boolean
}
